add paging feature.

paulsheldrake opened the pull request:
- Automatically get the remaining pages of the results if they are available.
- Refactors rate limit checking to make it simpler

This merge targets v1.4.0 as that is what we're using in our project but can update when a new version is released that matches the primary branch

paulsheldrake committed:
add paging feature.

checks if there are more pages available in the request and automatically fetches them.

simonschaufi commented:
Thank you for your contribution. Can you please rebase your branch from origin/master?

paulsheldrake committed:
Update Rest.php

Header info is now directly accessible and we don't need to access through the first key [0]
